File: Python/Problems/Problems/problem1968.py
from .problem import Problem


class Problem1968(Problem):
    def test(self):
        nums = [1, 2, 3, 4, 5]

        expected = [1, 3, 2, 5, 4]

        result = self.rearrangeArray(nums)

        return expected == result

    def rearrangeArray(self, nums):
        for i in range(1, len(nums) - 1):
            if (nums[i - 1] < nums[i] < nums[i + 1]) or (nums[i - 1] > nums[i] > nums[i + 1]):
                (nums[i], nums[i + 1]) = (nums[i + 1], nums[i])

        return nums

File: C++/Problems/Problems/problem_1968.cpp
#include "problem_1968.h"

bool problem_1968::test() {
    std::vector nums { 1, 2, 3, 4, 5 };

    std::vector expected { 1, 3, 2, 5, 4 };

    auto result = rearrangeArray(nums);

    return expected == result;
}

std::vector<int> problem_1968::rearrangeArray(std::vector<int> &nums) {
    for (int i = 1; i < nums.size() - 1; i++)
    {
        if ((nums[i - 1] < nums[i] && nums[i] < nums[i + 1])
            || (nums[i - 1] > nums[i] && nums[i] > nums[i + 1]))
            std::swap(nums[i], nums[i + 1]);
    }

    return nums;
}
